engine/src/main/java/org/hibernate/search/engine/metadata/impl/AnnotationMetadataProvider.java

```diff
@@ -427,37 +430,57 @@ private Annotation getIdAnnotation(XProperty member, TypeMetadata.Builder typeMe
 		return idAnnotation;
 	}
 
-	private void initializeProvidedIdMetadata(ProvidedId providedId, XClass clazz, TypeMetadata.Builder typeMetadataBuilder) {
+	private void initializeProvidedIdMetadata(String prefix, ProvidedId providedId, XClass clazz, TypeMetadata.Builder typeMetadataBuilder,
+			boolean isRoot, PathsContext pathsContext, ParseContext parseContext) {
 		PropertyMetadata propertyMetadata;
-		FieldBridge providedIdFieldBridge;
-		String providedIdFieldName;
+		FieldBridge providedIdFieldBridge = null;
+		String providedIdFieldPath;
 		if ( providedId != null ) {
-			providedIdFieldBridge = bridgeFactory.extractTwoWayType( providedId.bridge(), clazz, reflectionManager );
-			providedIdFieldName = providedId.name();
+			providedIdFieldPath = prefix + providedId.name();
 		}
 		else {
-			providedIdFieldBridge = new TwoWayString2FieldBridgeAdaptor( org.hibernate.search.bridge.builtin.StringBridge.INSTANCE );
-			providedIdFieldName = ProvidedId.defaultFieldName;
+			providedIdFieldPath = prefix + ProvidedId.defaultFieldName;
+		}
+
+		if ( !parseContext.includeEmbeddedObjectId() && pathsContext != null && !pathsContext.containsPath( providedIdFieldPath ) ) {
+			return;
+		}
+
+		if ( isRoot || !parseContext.skipFieldBridges() ) {
+			if ( providedId != null ) {
+				providedIdFieldBridge = bridgeFactory.extractTwoWayType( providedId.bridge(), clazz, reflectionManager );
+			}
+			else {
+				providedIdFieldBridge = new TwoWayString2FieldBridgeAdaptor( org.hibernate.search.bridge.builtin.StringBridge.INSTANCE );
+			}
 		}
 
 		PropertyMetadata.Builder propertyMetadataBuilder = new PropertyMetadata.Builder( typeMetadataBuilder.getResultReference(), null, null );
 
-		DocumentFieldMetadata fieldMetadata =
+		DocumentFieldMetadata.Builder fieldMetadataBuilder =
 				new DocumentFieldMetadata.Builder(
 						typeMetadataBuilder.getResultReference(),
 						propertyMetadataBuilder.getResultReference(),
-						providedIdFieldName,
+						providedIdFieldPath,
 						Store.YES,
 						Field.Index.NOT_ANALYZED_NO_NORMS,
 						Field.TermVector.NO
 				)
 						.fieldBridge( providedIdFieldBridge )
-						.boost( 1.0f )
-						.build();
+						.boost( 1.0f );
+
+		if ( !isRoot ) {
+			fieldMetadataBuilder.idInEmbedded();
+		}
+
+		DocumentFieldMetadata fieldMetadata = fieldMetadataBuilder.build();
 
 		propertyMetadataBuilder.addDocumentField( fieldMetadata );
 		propertyMetadata = propertyMetadataBuilder.build();
-		typeMetadataBuilder.idProperty( propertyMetadata );
+
+		if ( isRoot ) {
+			typeMetadataBuilder.idProperty( propertyMetadata );
```

Comment:
Mmmmh, did I miss something or if `!isRoot`, the whole work we did above was totally useless?